nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -199,12 +201,12 @@ void ObstacleLayer::onInitialize()
           *sub, *tf_, global_frame_, 50, rclcpp_node_));
 
       if (inf_is_valid) {
-        filter->registerCallback(std::bind(
+        connection_ = filter->registerCallback(std::bind(
```

Comment:
I'm confused by this, it looks like you're overwriting this variable each time its called, I don't believe that's having the desired effect.

Recall that the Obstacle Layer can take in N sensor inputs. Typically I dump all my 3D cameras into 1 layer for performance reasons. This would only shut down the last loaded connection. 

I'm not entirely sure why its necessary, can you add a quick explanation of what this fixes in the PR for posterity?

Reply:
The thought was that the registerCallback method is returning a connection object that needs to be disconnected. I didn't want race conditions at shutdown. I see now that you point it out that we could take multiple sensor inputs of the same kind. So, I've deleted the connection_ member for now.